CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/SplitNodesPlugin.java

```diff
@@ -157,19 +157,16 @@ private void editResultStore(final RecordStore result, final String left, final
         result.set(GraphRecordStoreUtilities.DESTINATION + VisualConcept.VertexAttribute.IDENTIFIER, right);
 
         //Loops through all of the Node attributes and copies them to the new node
-        query.reset();
-        while (query.next()) {
-            for (final String key : query.keys()) {
-                if (key.endsWith(".[id]") || SOURCE_IDENTIFIER.equals(key)) {
-                    //Skips the id and Identifier to make the new node unique
-                } else if ((GraphRecordStoreUtilities.SOURCE + VisualConcept.VertexAttribute.X).equals(key)
-                        || (GraphRecordStoreUtilities.SOURCE + VisualConcept.VertexAttribute.Y).equals(key)
-                        || (GraphRecordStoreUtilities.SOURCE + VisualConcept.VertexAttribute.Z).equals(key)) {
-                    //The coordinates are also skipped so that the second node is not created in the exact same location
-                    //as the first node
-                } else {
-                    result.set(GraphRecordStoreUtilities.DESTINATION + key.replace(GraphRecordStoreUtilities.SOURCE, ""), query.get(key));
-                }
+        for (final String key : query.keys()) {
```

Comment:
I must be missing something here - but I would have thought that by removing the `RecordStore` iterator you're treating it as a `Record` and so only the one data record currently pointed at will be modified.

Reply:
The original bug was the the `query` RecordStore (passed by reference) in the this method was looping through to the end which meant that when you returned from `editRecordStore` you were at the end so all other nodes selected would not get a turn.

The intension of this bit of the code was to share the attributes of the node that is being split with the splitter nodes. Therefore we don't need to look and we only care about the keys of the current record.

I wrote a unit test to confirm that it works as expected.

Reply:
Ahh I see, the iteration is happening elsewhere.